### DISCUSSION TITLE
e2e

### DISCUSSION THREAD

**Discussion by gonzalo-smtv** [Web3Auth/Web3Auth#985] [Help]
Hey, I want to know how to implement e2e test for my dApp with Web3Auth

**Comment by metallicalfa2** [comment]
You can implement an e2e test with a custom verifier which uses your JWT server (Its hard to implement an e2e for google login). 
https://web3auth.io/docs/custom-authentication/byo-jwt-providers

**Reply by gonzalo-smtv** [reply]
Thanks!
